plugin.js

```diff
@@ -45,6 +47,162 @@ const globals = new Set([
   '_scrollTo',
 ]);
 
+const blacklistedFunctions = new Set([
+  'stopCapturing',
+  'toString',
+  'map',
+  'filter',
+  'forEach',
+  'valueOf',
+  'toPrecision',
+  'toExponential',
+  'constructor',
+  'toFixed',
+  'toLocaleString',
+  'toSource',
+  'charAt',
+  'charCodeAt',
+  'concat',
+  'indexOf',
+  'lastIndexOf',
+  'localeCompare',
+  'length',
+  'match',
+  'replace',
+  'search',
+  'slice',
+  'split',
+  'substr',
+  'substring',
+  'toLocaleLowerCase',
+  'toLocaleUpperCase',
+  'toLowerCase',
+  'toUpperCase',
+  'every',
+  'join',
+  'pop',
+  'push',
+  'reduce',
+  'reduceRight',
+  'reverse',
+  'shift',
+  'slice',
+  'some',
+  'sort',
+  'splice',
+  'unshift',
+  'hasOwnProperty',
+  'isPrototypeOf',
+  'propertyIsEnumerable',
+  'bind',
+  'apply',
+  'call',
+]);
+
+class ClosureGenerator {
+  trie = [{}, false];
+
+  mergeAns(oldAns, newAns) {
+    const [purePath, node] = oldAns;
+    const [purePathUp, nodeUp] = newAns;
+    if (purePathUp.length !== 0) {
+      return [purePath.concat(purePathUp), nodeUp];
+    } else {
+      return [purePath, node];
+    }
+  }
+
+  findPrefixRec(path) {
+    const notFound = [[], null];
+    if (!path || path.node.type !== 'MemberExpression') {
+      return notFound;
+    }
+    const memberExpressionNode = path.node;
+    if (memberExpressionNode.property.type !== 'Identifier') {
+      return notFound;
+    }
+    if (
+      memberExpressionNode.computed ||
+      memberExpressionNode.property.name === 'value' ||
+      blacklistedFunctions.has(memberExpressionNode.property.name)
+    ) {
+      // a.b[w] -> a.b.w in babel nodes
+      // a.v.value
+      // sth.map(() => )
```

Comment:
is this comment necessary?

Reply:
It explains the main edge case in which we don't want to capture the prototypes methods like map for example.